flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/FCU/A320_Neo_FCU.js

```diff
@@ -432,6 +432,24 @@ class A320_Neo_FCU_Speed extends A320_Neo_FCU_Component {
             this.onPull();
         } else if (_event === "SPEED_TOGGLE_SPEED_MACH") {
             this.onSwitchSpeedMach();
+        } else if (_event === "USE_PRE_SEL_SPEED") {
+            clearTimeout(this._resetSelectionTimeout);
+            SimVar.SetSimVarValue("K:SPEED_SLOT_INDEX_SET", "number", 1);
+            this.inSelection = false;
+            this.isSelectedValueActive = false;
+            this.isTargetManaged = false;
+            this.isMachActive = false;
+            this.selectedValue = SimVar.GetSimVarValue("L:A32NX_SpeedPreselVal", "knots");
+            SimVar.SetSimVarValue("K:AP_MANAGED_SPEED_IN_MACH_OFF", "number", 1);
+        } else if (_event === "USE_PRE_SEL_MACH") {
+            clearTimeout(this._resetSelectionTimeout);
+            SimVar.SetSimVarValue("K:SPEED_SLOT_INDEX_SET", "number", 1);
+            this.inSelection = false;
+            this.isSelectedValueActive = false;
+            this.isTargetManaged = false;
+            this.isMachActive = true;
+            this.selectedValue = SimVar.GetSimVarValue("L:A32NX_MachPreselVal", "mach");
+            SimVar.SetSimVarValue("K:AP_MANAGED_SPEED_IN_MACH_ON", "number", 1);
```

Comment:
Maybe better to factor these two into a function like the existing ones, like `onPreSelSpeed(mach)`, so a) you can de-duplicate most of the code, and b) to keep the flow simple and more readable in this function.